tests/test_discriminated_union.py

```diff
@@ -1763,3 +1763,90 @@ class CreateObjectDto(BaseModel):
             'type': 'object',
         },
     }
+
+
+def test_nested_discriminator() -> None:
+    """
+    The exact details of the JSON schema produced are not necessarily important; the test was added in response to a
+    regression that caused the inner union was losing its discriminator. Even if the schema changes, the important
+    thing is that the core schema (and therefore JSON schema) produced has an actual discriminated union in it.
+    """
+
+    class Step_A(BaseModel):
+        type: Literal['stepA']
+        count: int
+
+    class Step_B(BaseModel):
+        type: Literal['stepB']
+        value: float
+
+    class MyModel(BaseModel):
+        type: Literal['mixed']
+        sub_models: list['SubModel']
```

Comment:
```suggestion
        sub_models: List['SubModel']
```

Will have to change this to make the 3.8 tests happy